packages/core/core/src/types.js

```diff
@@ -57,6 +57,7 @@ export type Dependency = {|
   isOptional: boolean,
   isURL: boolean,
   isWeak: ?boolean,
+  isDeferred: boolean, // TODO, only meaningful in bundlegraph
```

Comment:
Should I create a `BundleGraphDependency(Node)` type, because `isDeferred` isn't meaningful in the AssetGraph (at that point, the connected `AssetGroupNode` has that information.